design/gw-api-policy-and-filter-design.md

```diff
@@ -0,0 +1,238 @@
+
+This potentially gives users a huge amount of flexibility *and* operators control over critical settings.
+We can concievably have a kubectl plugin or other visualization that shows exactly what settings are enabled on a particular route.
+
+Cross-namespace references also are possible with Policies, though not implemented in this example, but possible in the future with Policy resources accompanied by ReferenceGrants.
+
+Overall this option has more moving pieces, but is very powerful.
+In addition, it does not directly affect the core Gateway API resources, but rather is a meta-resource on top of them that adds additional functionality.
+This may be desirable for some.
+
+It is also worth noting, we don't necessarily have to allow a Policy to target all resources, we can constrain this and as mentioned in implementation notes, it might be a lot easier to iterate this way.
+
+#### Rate Limiting: Custom BackendObjectReference type
+This option involves wrapping a Service that requires a rate limit with a resource that specifies a rate limit.
+
+I'm not entirely sure this will properly work, as we can't today implement per-cluster rate limits in Envoy, though there may be a clever way in the API we could get it to work.
+
+Implementing features using this method (and with Filters to be fair) does mean we affect the core resources, making it a bit harder to swap one's configuration between Gateway API implementations.
+
+### Idea: Replace parts of Contour global configuration with Policies
+An option for simplifying the ContourConfiguration and ContourDeployment CRDs could be to split some of the relevant fields it contains out into Policy resources that are applicable to a Gateway.
+
+This way, there is a more succinct and focused place for users to apply configuration, and we can acheive some goals of having more dynamic configuration for some Contour settings.
+The configuration file/configuration CRDs should end up just being what is needed to get Contour/Envoy started.
+
+The Gateway Provisioner could give you a base/default Policy if none existed or users could come with their own.
+
+Thinking out loud, this might alleviate some of the weirdness around how GatewayClass parameters maybe shouldn't be reconciled when things change.
+
+Some examples of things that might be applicable here to remove from the global config and move to a Gateway policy are:
+- Timeout policy
+- HTTP header policy
+- TLS parameters
+- Envoy "Cluster" settings
+- Envoy "Listener" settings
+- etc.
+
+## Security Considerations
+
+### Cross-namespace references
+It is not specified explicitly in the Gateway API documentation, but if Policy or Filter custom resources live in a separate namespace from the resource it references/is referenced by, we will likely need to ensure the appropriate ReferenceGrant is present that allows the resources to be "attached."
```

Comment:
[ExtensionRef](https://github.com/kubernetes-sigs/gateway-api/blob/v0.5.1/apis/v1beta1/httproute_types.go#L601-L609) is a LocalObjectRef so the ref'd resource should be in the same ns.

Reply:
ah yep, good catch, should be corrected to match the `Rate Limiting: Filter` section which mentions this